libs/sdk-ui-kit/src/Dialog/ShareDialog/ShareDialogBase/GranularPermissions/GranularGranteeUserItem.tsx

```diff
@@ -1,31 +1,51 @@
 // (C) 2022-2023 GoodData Corporation
 
-import React, { useMemo } from "react";
+import React, { useCallback, useMemo } from "react";
 import { useIntl } from "react-intl";
 import cx from "classnames";
 
 import { GranteeUserIcon } from "../GranteeIcons";
-import { GranteeItem, IGranularGranteeUser } from "../types";
+import { DialogModeType, GranteeItem, IGranularGranteeUser } from "../types";
 import { getGranteeItemTestId, getGranteeLabel } from "../utils";
 import { CurrentUserPermissions } from "../../types";
 
 import { GranularPermissionsDropdownWithBubble } from "./GranularPermissionsDropdown";
 import { usePermissionsDropdownState } from "./usePermissionsDropdownState";
 import { getGranteePossibilities } from "./permissionsLogic";
+import { useComponentInteractionContext } from "../ComponentInteractionContext";
 
 interface IGranularGranteeUserItemProps {
     grantee: IGranularGranteeUser;
     currentUserPermissions: CurrentUserPermissions;
     isDashboardLocked: boolean;
+    mode: DialogModeType;
     onChange: (grantee: GranteeItem) => void;
     onDelete: (grantee: GranteeItem) => void;
 }
 
 export const GranularGranteeUserItem: React.FC<IGranularGranteeUserItemProps> = (props) => {
-    const { grantee, currentUserPermissions, isDashboardLocked, onChange, onDelete } = props;
-    const { email } = grantee;
+    const { grantee, currentUserPermissions, isDashboardLocked, onChange, onDelete, mode } = props;
+    const { email, permissions } = grantee;
     const { isDropdownOpen, toggleDropdown } = usePermissionsDropdownState();
+    const onInteraction = useComponentInteractionContext();
     const intl = useIntl();
+
+    const handleToggleDropdown = useCallback(() => {
```

Comment:
Could you extract this logic from component ? Separate hook? or util function?. And move this closer to permissions ?

Reply:
done